sdks/java/core/src/test/java/org/apache/beam/sdk/io/TFRecordSchemaTransformProviderTest.java

```diff
@@ -0,0 +1,609 @@
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.collect.Sets;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.io.BaseEncoding;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.io.ByteStreams;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for TFRecordIO Read and Write transforms. */
+@RunWith(JUnit4.class)
+public class TFRecordSchemaTransformProviderTest {
+
+  /*
+  From https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/tfrecordio_test.py
+  Created by running following code in python:
+  >>> import tensorflow as tf
+  >>> import base64
+  >>> writer = tf.python_io.TFRecordWriter('/tmp/python_foo.tfrecord')
+  >>> writer.write('foo')
+  >>> writer.close()
+  >>> with open('/tmp/python_foo.tfrecord', 'rb') as f:
+  ...   data = base64.b64encode(f.read())
+  ...   print data
+  */
+  private static final String FOO_RECORD_BASE64 = "AwAAAAAAAACwmUkOZm9vYYq+/g==";
+
+  // Same as above but containing two records ['foo', 'bar']
+  private static final String FOO_BAR_RECORD_BASE64 =
+      "AwAAAAAAAACwmUkOZm9vYYq+/gMAAAAAAAAAsJlJDmJhckYA5cg=";
+  private static final String BAR_FOO_RECORD_BASE64 =
+      "AwAAAAAAAACwmUkOYmFyRgDlyAMAAAAAAAAAsJlJDmZvb2GKvv4=";
+
+  private static final String[] FOO_RECORDS = {"foo"};
+  private static final String[] FOO_BAR_RECORDS = {"foo", "bar"};
+
+  private static final Iterable<String> EMPTY = Collections.emptyList();
+  private static final Iterable<String> SMALL = makeLines(1, 4);
+  private static final Iterable<String> LARGE = makeLines(1000, 4);
+  private static final Iterable<String> LARGE_RECORDS = makeLines(100, 100000);
+
+  @Rule public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  @Rule public TestPipeline readPipeline = TestPipeline.create();
+
+  @Rule public TestPipeline writePipeline = TestPipeline.create();
+
+  @Rule public ExpectedException expectedException = ExpectedException.none();
+
+  private static TFRecordReadSchemaTransformConfiguration readConfig;
+  private static TFRecordWriteSchemaTransformConfiguration writeConfig;
+
+  @Before
+  public void setUp() throws IOException {
+    readConfig =
+        TFRecordReadSchemaTransformConfiguration.builder()
+            .setValidate(false)
+            .setCompression("AUTO")
+            .setFilePattern("foo.*")
+            .build();
+
+    writeConfig =
+        TFRecordWriteSchemaTransformConfiguration.builder()
+            .setOutputPrefix(tempFolder.getRoot().toPath().toString())
+            .setFilenameSuffix("bar")
+            .setShardTemplate("xyz")
+            .setNumShards(10)
+            .setCompression("UNCOMPRESSED")
+            .setNoSpilling(true)
+            .build();
```

Comment:
nit: i think you can initialize these inline?

Reply:
I had intentions of using this in more places, but it didn't work out.  Will inline as suggested.  Thanks